### PR TITLE
Status bar messages sourced from persist.raw_errors.

### PR DESCRIPTION
**Pull request by pykong**
Earliest working version

For @kaste with <3

### CODE REVIEW

**Comment by kaste** [comment]
This looks promising. I see you get a feeling for data transformations in python and why python has all these comprehension syntax forms.

**Comment by pykong** [comment]
It just bugs me I created this convoluted `errors` monstrosity. 
Again it is amazing how much code just evaporates by changing this core data structure.

**Comment by braver** [comment]
LGTM 👍🏻

**Comment by kaste** [comment]
Updated title. This is [WIP]

**Comment by pykong** [comment]
@braver Did you test this branch? LGTM. Can we move forward with this PR, so the way gets cleared for #822?

**Comment by braver** [comment]
Is this not still WIP then?

**Comment by kaste** [comment]
`get_we_count` must be re-implemented

**Comment by kaste** [comment]
This is blocked by #822 

`get_we_count` implementation is trivial

**Comment by braver** [comment]
What is blocking what now? If they're so intertwined, shouldn't we merge it into a single PR?

**Comment by kaste** [comment]
Merge order is #822 then #851 

They're only intertwined bc status_bar imports and controls the panel, if you remember that.

**Comment by kaste** [comment]
@pykong here is fine, we add like 5 lines here, and done.

**Comment by braver** [comment]
Okido. @pykong the ball is in your court then.

**Comment by kaste** [comment]
LGTM 👍

File: status_bar_view.py [hunk 1]
```diff
@@ -58,30 +68,45 @@ def on_selection_modified_async(self, _primary_view_):
...
+def messages_under_cursor(errors, current_pos):
+    line, col = current_pos
+    return [
+        error['msg'] for error in errors[line]
+        if error["start"] <= col <= error["end"]
+    ]
+
+
+def errors_per_line(errors):
+    rv = defaultdict(list)
```
Comment (braver) [inline status_bar_view.py]: what's an rv?

3689-jayco2017_precept31ul_34front_oceanblue

🤔

Comment (kaste) [reply status_bar_view.py]: Hm, that's the common name for `return value`?

Comment (kaste) [reply status_bar_view.py]: Done

Comment (braver) [reply status_bar_view.py]: > Hm, that's the common name for return value?

I guessed that but haven't seen it used like that much. It's all good 👍

**Comment by braver** [comment]
LGTM from me too then!

File: status_bar_view.py (hunk 1 above)
Comment (FichteFoll) [reply status_bar_view.py]: Interesting. I always use `ret` or `ret_val` for that.

Comment (kaste) [reply status_bar_view.py]: I think I have this from Armin Ronacher. ... yeaaars ago

**Pull request merged**